core/payment-driver/src/gnt/sender.rs

```diff
@@ -689,3 +766,155 @@ impl TransactionSender {
+        ActorResponse::reply(Ok(()))
+    }
+}
+
+async fn process_payments(
+    account: String,
+    client: Arc<EthereumClient>,
+    gnt_contract: Arc<Contract<Http>>,
+    tx_sender: Addr<TransactionSender>,
+    db: DbExecutor,
+    sign_tx: SignTx<'_>,
+) {
+    match db
+        .as_dao::<PaymentDao>()
+        .get_pending_payments(account.clone())
+        .await
+    {
+        Err(e) => log::error!(
+            "Failed to fetch pending payments for {:?} : {:?}",
+            account,
+            e
+        ),
+        Ok(payments) => {
+            for payment in payments {
+                let _ = process_payment(
+                    payment.clone(),
+                    client.clone(),
+                    gnt_contract.clone(),
+                    tx_sender.clone(),
+                    db.clone(),
+                    sign_tx,
+                )
+                .await
+                .map_err(|e| {
+                    log::error!("Failed to process payment: {:?}, error: {:?}", payment, e)
+                });
+            }
+        }
+    };
+}
+
+async fn process_payment(
+    payment: PaymentEntity,
+    client: Arc<EthereumClient>,
+    gnt_contract: Arc<Contract<Http>>,
+    tx_sender: Addr<TransactionSender>,
+    db: DbExecutor,
+    sign_tx: SignTx<'_>,
+) -> PaymentDriverResult<()> {
+    log::info!("Processing payment: {:?}", payment);
+    let gas_price = client.get_gas_price().await?;
+    let chain_id = client.chain_id();
+    match transfer_gnt(
+        gnt_contract,
+        tx_sender,
+        utils::u256_from_big_endian_hex(payment.amount),
+        utils::str_to_addr(&payment.sender)?,
+        utils::str_to_addr(&payment.recipient)?,
+        sign_tx,
+        gas_price,
+        chain_id,
+    )
+    .await
+    {
+        Ok(tx_id) => {
+            db.as_dao::<PaymentDao>()
+                .update_tx_id(payment.invoice_id, tx_id)
+                .await?;
+        }
+        Err(e) => {
+            db.as_dao::<PaymentDao>()
+                .update_status(
+                    payment.invoice_id,
+                    match e {
+                        PaymentDriverError::InsufficientFunds => PAYMENT_STATUS_NOT_ENOUGH_FUNDS,
+                        PaymentDriverError::InsufficientGas => PAYMENT_STATUS_NOT_ENOUGH_GAS,
+                        _ => PAYMENT_STATUS_FAILED,
+                    },
+                )
+                .await?;
+            log::error!("GNT transfer failed: {}", e);
+            return Err(e);
+        }
+    }
+    Ok(())
+}
+
+async fn transfer_gnt(
+    gnt_contract: Arc<Contract<Http>>,
+    tx_sender: Addr<TransactionSender>,
+    gnt_amount: U256,
+    address: Address,
+    recipient: Address,
+    sign_tx: SignTx<'_>,
+    gas_price: U256,
+    chain_id: u64,
+) -> PaymentDriverResult<String> {
+    if gnt_amount
+        > utils::big_dec_to_u256(
+            common::get_gnt_balance(&gnt_contract, address)
+                .await?
+                .amount,
```

Comment:
This if condition is hardly readable due to the formatting. Maybe assign the gnt balance to a variable?

Reply:
Ok